APIKit/APIKit.swift

```diff
@@ -119,4 +192,27 @@ public class API {
             return nil
         }
     }
+    
+    // MARK: - NSURLSessionDelegate
+    public func URLSession(session: NSURLSession, didReceiveChallenge challenge: NSURLAuthenticationChallenge, completionHandler: (NSURLSessionAuthChallengeDisposition, NSURLCredential!) -> Void) {
```

Comment:
This is an optional delegate method. Should we need to set up default implementation?